packages/actionmenu/src/react/item.tsx

```diff
@@ -43,99 +47,107 @@ const styles = {
+  disabled?: boolean
+  href?: string
   nested?: React.ReactNode
+  onClick?: (event: React.MouseEvent, value: string | number) => void
+  origin?: string
+  tagName?: string
   value?: string | number
 }
-
-export const Item = forwardRef<HTMLLIElement, Props>(
-  (
-    {
+export const Item = forwardRef<HTMLLIElement, ItemProps>(
+  (props, forwardedRef) => {
+    const {
+      children,
+      className,
       disabled,
       nested,
+      onClick,
       origin,
+      tagName: TagName = tagName.a,
       value,
-      className,
-      children,
-      onClick,
-      tagName,
       ...rest
-    },
-    forwardedRef
-  ) => {
-    const { onClickContext, onClose, originContext } = useContext<{
-      onClickContext: (e: Event, v: string | number) => void
-      onClose: (e: Event, v: string | number) => void
-      originContext: string
-    }>(ActionMenuContext)
+    } = props
+
+    const { onClickContext, onClose, originContext } = useContext(
+      ActionMenuContext
+    )
+
     const ref = useRef()
-    const subMenuRef = useRef()
     useImperativeHandle(forwardedRef, () => ref.current)
-    const TagName = tagName
+
+    const subMenuRef = useRef()
+
     const [open, setOpen] = useState(false)
     const hasSubMenu = Boolean(nested)
-    const handleMouseOver = e => {
+
+    const handleMouseOver: MouseEventHandler<HTMLLIElement> = () => {
       hasSubMenu && setOpen(true)
     }
-    const handleMouseOut = e => {
+
+    const handleMouseOut: MouseEventHandler<HTMLLIElement> = () => {
       hasSubMenu && setOpen(false)
     }
-    const handleKeyDown = e => {
-      if (e.key === 'ArrowRight' && hasSubMenu) {
+
+    const handleKeyDown: KeyboardEventHandler<HTMLLIElement> = evt => {
+      if (evt.key === 'ArrowRight' && hasSubMenu) {
         setOpen(true)
-        e.stopPropagation()
+        evt.stopPropagation()
       }
-      if (e.key === 'Enter' || e.key === 'Space') {
-        e.target.firstElementChild.click()
+
+      if (evt.key === 'Enter' || evt.key === 'Space') {
+        const target = evt.target as HTMLUListElement
+        const firstEl = target.firstElementChild as HTMLLIElement
+        firstEl.click()
       }
-      e.preventDefault()
+
+      evt.preventDefault()
     }
-    const handleArrowLeft = e => {
-      if (e.key === 'ArrowLeft' && hasSubMenu) {
+    const handleArrowLeft: KeyboardEventHandler<HTMLUListElement> = evt => {
+      if (evt.key === 'ArrowLeft' && hasSubMenu) {
         setOpen(false)
-        e.stopPropagation()
+        evt.stopPropagation()
       }
     }
-    const handleClick = e => {
-      onClick && onClick(e, value)
-      onClickContext && onClickContext(e, value)
-      onClose && onClose(e, value) // : e.currentTarget.parentNode.focus()
+    const handleClick: MouseEventHandler<HTMLLIElement> = evt => {
+      onClick && onClick(evt, value)
+      onClickContext && onClickContext(evt, value)
+      onClose && onClose(evt, value) // : e.currentTarget.parentNode.focus()
```

Comment:
i noticed this comment....did we miss something that we need to fix?

Reply:
Not sure. Existed before 574702192b, the TS conversion.

Line was added, including comment, in 0248ef32d4 by @EdwardIrby  in a large component rebuild.  Might want to ask him.

Reply:
I modded this a bunch in Dropdown conversion so I'll just have to resolve the conflict later. I prever going with `evt: ReactMouiseEvt, value: React.ReactText`